### PR TITLE
Floating label tweaks

### PR DESCRIPTION
**Pull request by BrigadeTom**
screenshot_1505950606

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -6,7 +6,7 @@ buildscript {
         mavenCentral()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.0.0-beta5'
+        classpath 'com.android.tools.build:gradle:3.0.0-beta6'
```
Comment (ShengLiu) [inline build.gradle]: You may want to rebase this branch of the latest `master`. `master` branch should already included this change.

File: library/src/main/res/values/colors.xml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <color name="gray_3">#CECFD4</color>
```
Comment (brigade-juni) [inline library/src/main/res/values/colors.xml]: This seems like we're committing to not merging this back to MaterialEditText; that's ok though, right?

File: library/src/main/res/values/dimens.xml [hunk 3]
```diff
@@ -6,11 +6,13 @@
   <dimen name="inner_padding_right">0dp</dimen>
   <dimen name="floating_label_text_size">12sp</dimen>
   <dimen name="bottom_text_size">12sp</dimen>
-  <dimen name="inner_components_spacing">4dp</dimen>
+  <dimen name="inner_components_spacing">8dp</dimen>
```
Comment (ShengLiu) [inline library/src/main/res/values/dimens.xml]: question - where is this being used?

File: library/src/main/java/com/rengwuxian/materialedittext/MaterialEditText.java [hunk 4]
```diff
@@ -520,8 +534,9 @@ private void initTextArea() {
         @Override
         public void onFocusChange(final View v, final boolean hasFocus) {
           if (hasFocus) {
-            setOnTouchListener(new OnTouchListener() {
+            setBackground(focusStateRectBorderLayerDrawable);
```
Comment (brigade-juni) [inline library/src/main/java/com/rengwuxian/materialedittext/MaterialEditText.java]: seems like we should be using a state list drawable for the background

Comment (BrigadeTom) [reply library/src/main/java/com/rengwuxian/materialedittext/MaterialEditText.java]: made the change. Had to create the StateListDrawable in code as I need to make runtime adjustments to the drawables used for the different states.

File: library/src/main/res/values/dimens.xml (hunk 3 above)
Comment (BrigadeTom) [reply library/src/main/res/values/dimens.xml]: In the MaterialTextEdit. Defines the distance between the underline and counter + error text.

Comment (ShengLiu) [reply library/src/main/res/values/dimens.xml]: 👍

File: library/src/main/res/values/colors.xml (hunk 2 above)
Comment (ShengLiu) [reply library/src/main/res/values/colors.xml]: maybe we should give it a more generic and not brigade specific name. Something like `disabledColor` or `defaultColor`.

**Pull request merged**